### PR TITLE
[Internal] Refactoring: Refactors Moving to ValueFactory overloads of ConcurrentDictionary

### PR DESCRIPTION
**Pull request by kirankumarkolli**
Moving to ValueFactory overloads of ConcurrentDictionary

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs [hunk 1]
```diff
@@ -571,7 +571,8 @@ await this.GetServerAddressesViaGatewayAsync(request, collectionRid, new[] { par
                        addressInfo.PhysicalUri);
 
                     HashSet<PartitionKeyRangeIdentity> pkRangeIdSet = this.serverPartitionAddressToPkRangeIdMap.GetOrAdd(
-                        new ServerKey(new Uri(addressInfo.PhysicalUri)), new HashSet<PartitionKeyRangeIdentity>());
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs:574]: I don't think there is any way to test this, is it?

**Review by ealsur — APPROVED** [review]

**Comment by ealsur** [comment]
Why is it marked as [Internal] ?

File: Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs (hunk 1 above)
Comment (j82w) [reply Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs:574]: The contract could be changed to always take in a lambda to force it to always be lazy.

**Comment by bartelink** [comment]
nit: typo in Title: Refractoring

**Comment by kirankumarkolli** [comment]
> Why is it marked as [Internal] ?

No user visible feature. Its a very minor performance impact.

File: Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs (hunk 1 above)
Comment (kirankumarkolli) [reply Microsoft.Azure.Cosmos/src/Routing/GatewayAddressCache.cs:574]: Functional: Existing tests cover it. 
Performance: May be some GC stats with running it E2E might help. 

Future: Ideal is to explore static analysis tool rule to block this API usage from Concurrent collections.

**Pull request merged**